pages/accountLists/[accountListId]/contacts/flows/setup.page.tsx

```diff
@@ -138,27 +88,35 @@ const ContactFlowSetupPage: React.FC = () => {
 
   const changeColor = (index: number, color: string): void => {
     const temp = [...flowOptions];
-    temp[index].color = color;
+    temp[index] = { ...temp[index], color };
     updateOptions(temp);
   };
 
-  const moveStatus = (
-    originIndex: number,
-    destinationIndex: number,
-    draggedStatus: StatusEnum,
-  ): void => {
-    const temp = [...flowOptions];
-    if (originIndex > -1) {
-      temp[originIndex].statuses = temp[originIndex].statuses.filter(
-        (status) => status !== draggedStatus,
-      );
-    }
-    if (destinationIndex > -1) {
-      temp[destinationIndex].statuses.push(draggedStatus);
-    }
-    updateOptions(temp);
-    setFlowOptions(temp);
-  };
+  const moveStatus = useCallback(
+    (
+      originIndex: number,
+      destinationIndex: number,
+      draggedStatus: StatusEnum,
+    ): void => {
+      const temp = [...flowOptions];
+      if (originIndex > -1) {
+        temp[originIndex] = {
+          ...temp[originIndex],
+          statuses: temp[originIndex].statuses.filter(
+            (status) => status !== draggedStatus,
+          ),
+        };
+      }
+      if (destinationIndex > -1) {
+        temp[destinationIndex] = {
+          ...temp[destinationIndex],
+          statuses: [...temp[destinationIndex].statuses, draggedStatus],
+        };
+      }
```

Comment:
Before this was updating cloning the array of flow options but still modifying the original column. This clones the column too before updating it.